sources/packages/backend/apps/api/src/services/supporting-user/supporting-user.service.ts

```diff
@@ -28,90 +26,48 @@ export class SupportingUserService extends RecordDataModelService<SupportingUser
-      // two parents tries to update the record at the same time the second one will wait
-      // till the first one finishes it.
-      const transactionRepo = queryRunner.manager.getRepository(SupportingUser);
-      const possibleUsersToUpdate = await transactionRepo
-        .createQueryBuilder("supportingUser")
-        .setLock("pessimistic_write")
-        .select("supportingUser.id")
-        .innerJoin("supportingUser.application", "application")
-        .where("supportingUser.supportingUserType = :supportingUserType", {
-          supportingUserType,
-        })
-        .andWhere("supportingUser.user.id is null")
-        .andWhere("application.id = :applicationId", { applicationId })
-        .orderBy("supportingUser.id")
-        .getMany();
-
-      // Case there are more than one records (e.g. two parents), just selected one
-      // to execute the update.
-      const userToUpdate = possibleUsersToUpdate.shift();
-      if (!userToUpdate) {
-        throw new CustomNamedError(
-          `The application is not expecting supporting information from a ${supportingUserType.toLowerCase()} to be provided at this time.`,
-          SUPPORTING_USER_TYPE_ALREADY_PROVIDED_DATA,
-        );
-      }
+  ): Promise<UpdateResult> {
+    const sinWithNoSpaces = updateInfo.sin
+      ? removeWhiteSpaces(updateInfo.sin)
+      : null;
+    const personalInfo = updateInfo.hasValidSIN
+      ? { hasValidSIN: updateInfo.hasValidSIN }
+      : null;
 
-      // The SIN can be received with a few white spaces in the middle, so we need remove then
-      // before inserting it because the DB will accept only 9 characters.
-      const sinWithNoSpaces = updateInfo.sin
-        ? removeWhiteSpaces(updateInfo.sin)
-        : null;
-      userToUpdate.contactInfo = updateInfo.contactInfo;
-      userToUpdate.sin = sinWithNoSpaces;
-      userToUpdate.birthDate = updateInfo.birthDate;
-      userToUpdate.supportingData = updateInfo.supportingData;
-      userToUpdate.user = { id: updateInfo.userId } as User;
-      userToUpdate.modifier = { id: auditUserId } as User;
-      userToUpdate.personalInfo = updateInfo.hasValidSIN
-        ? { hasValidSIN: updateInfo.hasValidSIN }
-        : null;
-      const updatedUser = await transactionRepo.save(userToUpdate);
-      await queryRunner.commitTransaction();
-      return updatedUser;
-    } catch (error) {
-      await queryRunner.rollbackTransaction();
-      throw error;
-    } finally {
-      await queryRunner.release();
+    const updateResult = await this.repo.update(
+      { id: supportingUserId, user: { id: IsNull() } },
+      {
+        contactInfo: updateInfo.contactInfo,
+        sin: sinWithNoSpaces,
+        birthDate: updateInfo.birthDate,
+        supportingData: updateInfo.supportingData,
+        user: { id: updateInfo.userId } as User,
+        personalInfo,
+        modifier: { id: auditUserId } as User,
```

Comment:
To simplify we can declare an audit user variable and use it. 

`const auditUser = { id: auditUserId } as User;`